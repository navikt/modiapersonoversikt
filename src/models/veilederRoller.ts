export interface VeilederRoller {
    roller: string[];
}
