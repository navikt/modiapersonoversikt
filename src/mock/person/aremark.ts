import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { BostatusTyper, Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';
import { SikkerhetstiltakTyper } from '../../models/sikkerhetstiltak';
import { Personadresse } from '../../models/personadresse';

export const gateAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2017-11-25T21:29:14.049Z',
        sistEndretAv: 'SKD'
    },
    gateadresse: {
        gatenavn: 'Islandsgate',
        husnummer: '49',
        postnummer: '7693',
        poststed: 'Hovedøya'
    }
};

export const postboksAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    postboksadresse: {
        postboksnummer: '9698',
        postnummer: '7705',
        poststed: 'Færder fyr'
    }
};

export const matrikkelAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    matrikkeladresse: {
        postnummer: '5468',
        poststed: 'Lillomarka'
    }
};

export const utlandsAdresseAremark: Personadresse = {
    endringsinfo: {
        sistEndret: '2018-04-10T09:06:58.256Z',
        sistEndretAv: 'SKD'
    },
    utlandsadresse: {
        landkode: {
            kodeRef: 'SYD',
            beskrivelse: 'Syden'
        },
        adresselinjer: ['Ut av landet', 'Rundt svingen']
    }
};

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    folkeregistrertAdresse: gateAdresseAremark,
    alternativAdresse: postboksAdresseAremark,
    bankkonto: {
        banknavn: 'Bien sparebank',
        kontonummer: '12345678911',
        sistEndret: '2005-12-12',
        sistEndretAv: 'z9910',
        landkode: {
            kodeRef: 'NOR',
            beskrivelse: 'Norge'
        },
        valuta: {
            beskrivelse: 'Spesidaler',
            kodeRef: 'SPD'
        }
    },
    sikkerhetstiltak: {
        sikkerhetstiltaksbeskrivelse:
            'Får verken møte opp på NAV-kontor eller ringe på telefon ' + 'grunnet truende oppførsel og språk.',
        sikkerhetstiltakskode: SikkerhetstiltakTyper.FysiskOgTelefonUtestengelse,
        periode: {
            fra: '2005-12-12',
            til: '2025-12-12'
        }
    },
    navn: {
        endringsinfo: {
            sistEndret: '2012-12-12',
            sistEndretAv: 'NAV'
        },
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN'
    },
    tilrettelagtKomunikasjonsListe: [
        {
            kodeRef: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: { kodeRef: Diskresjonskoder.FORTROLIG_ADRESSE, beskrivelse: 'Sperret adresse, fortrolig' },
    statsborgerskap: { kodeRef: 'NOR', beskrivelse: 'NORGE' },
    personstatus: {
        dødsdato: undefined,
        bostatus: {
            kodeRef: BostatusTyper.Utvandret,
            beskrivelse: BostatusTyper.Utvandret
        }
    },
    sivilstand: {
        kodeRef: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [
        {
            rolle: Relasjonstype.Barn,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, fortrolig'
                },
                personstatus: {}
            }
        },
        {
            rolle: Relasjonstype.Barn,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.STRENGT_FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, strengt fortrolig'
                },
                personstatus: {}
            }
        },
        {
            harSammeBosted: true,
            rolle: Relasjonstype.Barn,
            tilPerson: {
                fødselsnummer: '10117100000',
                navn: {
                    sammensatt: 'Siri Bellomnavn Aremark',
                    fornavn: 'Siri',
                    mellomnavn: 'Bellomnavn',
                    etternavn: 'Aremark'
                },
                personstatus: {},
                alder: 15
            }
        },
        {
            harSammeBosted: false,
            rolle: Relasjonstype.Barn,
            tilPerson: {
                fødselsnummer: '10117100100',
                navn: {
                    sammensatt: 'Ola Ballomnavn Aremark',
                    fornavn: 'Ola',
                    mellomnavn: 'Ballomnavn',
                    etternavn: 'Aremark'
                },
                personstatus: {},
                alderMåneder: 5
            }
        },
        {
            harSammeBosted: false,
            rolle: Relasjonstype.Ektefelle,
            tilPerson: {
                navn: null,
                diskresjonskode: {
                    kodeRef: Diskresjonskoder.STRENGT_FORTROLIG_ADRESSE,
                    beskrivelse: 'Sperret adresse, strengt fortrolig'
                },
                personstatus: {}
            }
        }
    ],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            identifikator: '99887766',
            retningsnummer: { kodeRef: '+47', beskrivelse: 'Norge' },
            sistEndretAv: '10108000398'
        }
    }
};
