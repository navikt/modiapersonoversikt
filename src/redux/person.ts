import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';
import { getPerson } from '../api/person-api';
import { aremark } from '../mock/person-mock';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: aremark,
    status: STATUS.NOT_STARTED,
    error: null
};

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return getPerson(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}