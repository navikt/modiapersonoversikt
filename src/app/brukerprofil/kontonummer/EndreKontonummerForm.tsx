import * as React from 'react';
import { ChangeEvent, FormEvent } from 'react';
import { connect } from 'react-redux';

import Input from 'nav-frontend-skjema/lib/input';

import { STATUS } from '../../../redux/restReducers/utils';
import { Person } from '../../../models/person/person';
import { AppState } from '../../../redux/reducers';
import { VeilederRoller } from '../../../models/veilederRoller';
import Undertittel from 'nav-frontend-typografi/lib/undertittel';
import { FormKnapperWrapper } from '../BrukerprofilForm';
import KnappBase from 'nav-frontend-knapper';
import RadioPanelGruppe from 'nav-frontend-skjema/lib/radio-panel-gruppe';
import {
    EndreBankkontoState,
    erBrukersKontonummerUtenlandsk,
    formaterNorskKontonummer,
    hentEndringstekst,
    removeWhitespaceAndDot,
    tomBankkonto
} from './kontonummerUtils';
import UtenlandskKontonrInputs from './UtenlandskKontonummerInputs';
import RequestTilbakemelding from '../RequestTilbakemelding';
import { endreKontonummer, reset } from '../../../redux/restReducers/brukerprofil/endreKontonummer';
import { EndreKontonummerRequest } from '../../../redux/restReducers/brukerprofil/endreKontonummerRequest';
import { ignoreEnter } from '../utils/formUtils';
import { getValidNorskBankKontoForm, validerNorskBankKonto } from './norskKontoValidator';
import { ValideringsResultat } from '../../../utils/forms/FormValidator';
import { getValidUtenlandskKontoForm, validerUtenlandskKonto } from './utenlandskKontoValidator';
import EtikettGrå from '../../../components/EtikettGrå';
import { FormFieldSet } from '../../personside/visittkort/body/VisittkortStyles';
import { veilederHarPåkrevdRolleForEndreKontonummer } from '../utils/RollerUtils';
import { EndreKontonummerInfomeldingWrapper } from '../Infomelding';
import { reloadPerson } from '../../../redux/restReducers/personinformasjon';
import { loggEvent } from '../../../utils/frontendLogger';
import { AsyncDispatch } from '../../../redux/ThunkTypes';
import { Undertekst } from 'nav-frontend-typografi';
import { FormatertKontonummer } from '../../../utils/FormatertKontonummer';
import { erNyePersonoversikten } from '../../../utils/erNyPersonoversikt';
import KnappMedBekreftPopup from '../../../components/KnappMedBekreftPopup';

enum bankEnum {
    erNorsk = 'Kontonummer i Norge',
    erUtenlandsk = 'Kontonummer i utlandet'
}

interface State {
    bankkontoInput: EndreBankkontoState;
    bankkontoValidering: ValideringsResultat<EndreBankkontoState>;
    norskKontoRadio: boolean;
}

interface DispatchProps {
    reloadPerson: (fødselsnummer: string) => void;
    endreKontonummer: (fødselsnummer: string, request: EndreKontonummerRequest) => void;
    resetEndreKontonummerResource: () => void;
}

interface StateProps {
    resourceStatus: STATUS;
}

interface OwnProps {
    person: Person;
    veilederRoller: VeilederRoller;
}

type Props = DispatchProps & StateProps & OwnProps;

class EndreKontonummerForm extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = this.getInitialState();
        this.handleNorskKontonummerInputChange = this.handleNorskKontonummerInputChange.bind(this);
        this.updateBankkontoInputsState = this.updateBankkontoInputsState.bind(this);
        this.handleRadioChange = this.handleRadioChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.tilbakestill = this.tilbakestill.bind(this);
        this.slettKontonummer = this.slettKontonummer.bind(this);
    }

    componentWillUnmount() {
        this.resetResource();
    }

    componentDidUpdate(prevProps: Props) {
        if (this.kontonummerBleEndret(prevProps)) {
            this.props.reloadPerson(this.props.person.fødselsnummer);
        }
        const nyeDataPåPerson = prevProps.person !== this.props.person;
        if (nyeDataPåPerson) {
            this.setState(this.getInitialState());
        }
    }

    kontonummerBleEndret(prevProps: Props) {
        return prevProps.resourceStatus !== STATUS.SUCCESS && this.props.resourceStatus === STATUS.SUCCESS;
    }

    getInitialState(): State {
        const brukersBankkonto = this.getBrukersBankkonto();
        const erNorsk = !erBrukersKontonummerUtenlandsk(this.props.person);
        return {
            bankkontoInput: brukersBankkonto,
            norskKontoRadio: erNorsk,
            bankkontoValidering: erNorsk ? getValidNorskBankKontoForm() : getValidUtenlandskKontoForm()
        };
    }

    getBrukersBankkonto(): EndreBankkontoState {
        const person = this.props.person;
        if (!person.bankkonto) {
            return tomBankkonto;
        }
        return {
            ...tomBankkonto,
            ...person.bankkonto,
            norskKontonummer: erBrukersKontonummerUtenlandsk(person)
                ? ''
                : formaterNorskKontonummer(person.bankkonto.kontonummer),
            utenlandskKontonummer: erBrukersKontonummerUtenlandsk(person) ? person.bankkonto.kontonummer : ''
        };
    }

    slettKontonummer() {
        this.props.endreKontonummer(this.props.person.fødselsnummer, { kontonummer: '' });
    }

    handleSubmit(event: FormEvent<HTMLFormElement>) {
        event.preventDefault();
        const kontoInput = this.state.bankkontoInput;
        const valideringsResultat = this.state.norskKontoRadio
            ? validerNorskBankKonto(kontoInput)
            : validerUtenlandskKonto(kontoInput);

        if (!valideringsResultat.formErGyldig) {
            this.setState({
                bankkontoValidering: valideringsResultat
            });
            return;
        }

        const fnr = this.props.person.fødselsnummer;
        if (this.state.norskKontoRadio) {
            this.props.endreKontonummer(fnr, {
                kontonummer: removeWhitespaceAndDot(kontoInput.norskKontonummer)
            });
        } else {
            this.props.endreKontonummer(fnr, {
                kontonummer: removeWhitespaceAndDot(kontoInput.utenlandskKontonummer),
                landkode: kontoInput.landkode.kodeRef,
                valuta: kontoInput.valuta.kodeRef,
                swift: kontoInput.swift,
                banknavn: kontoInput.banknavn,
                bankkode: kontoInput.bankkode,
                bankadresse: kontoInput.adresse
            });
        }
        loggEvent('Endre kontonummer', 'Brukerprofil');
    }

    handleNorskKontonummerInputChange(event: ChangeEvent<HTMLInputElement>) {
        this.updateBankkontoInputsState({
            norskKontonummer: formaterNorskKontonummer(event.target.value)
        });
    }

    updateBankkontoInputsState(partial: Partial<EndreBankkontoState>) {
        const gyldigValidering = this.state.norskKontoRadio
            ? getValidNorskBankKontoForm()
            : getValidUtenlandskKontoForm();
        this.setState({
            bankkontoValidering: gyldigValidering,
            bankkontoInput: {
                ...this.state.bankkontoInput,
                ...partial
            }
        });
        this.resetResource();
    }

    handleRadioChange(event: React.SyntheticEvent<EventTarget>, value: string) {
        this.setState({
            norskKontoRadio: value === bankEnum.erNorsk
        });
    }

    tilbakestill() {
        this.props.resetEndreKontonummerResource();
        this.setState({
            ...this.getInitialState()
        });
    }

    formErEndret() {
        return JSON.stringify(this.getBrukersBankkonto()) !== JSON.stringify(this.state.bankkontoInput);
    }

    getNorskKontonrInputs() {
        return (
            <Input
                label="Kontonummer"
                value={this.state.bankkontoInput.norskKontonummer}
                onChange={this.handleNorskKontonummerInputChange}
                onKeyPress={ignoreEnter}
                feil={this.state.bankkontoValidering.felter.norskKontonummer.skjemafeil}
            />
        );
    }

    radioKnappProps() {
        return [
            { label: bankEnum.erNorsk, id: bankEnum.erNorsk, value: bankEnum.erNorsk },
            { label: bankEnum.erUtenlandsk, id: bankEnum.erUtenlandsk, value: bankEnum.erUtenlandsk }
        ];
    }

    resetResource() {
        if (this.props.resourceStatus !== STATUS.NOT_STARTED) {
            this.props.resetEndreKontonummerResource();
        }
    }

    kontonummerBleLagret() {
        return this.props.resourceStatus === STATUS.SUCCESS;
    }

    requestIsPending() {
        return this.props.resourceStatus === STATUS.LOADING;
    }

    render() {
        const sletteKnapp = this.props.person.bankkonto &&
            this.props.person.bankkonto.kontonummer &&
            !this.requestIsPending() && (
                <FormKnapperWrapper>
                    <KnappMedBekreftPopup
                        onBekreft={this.slettKontonummer}
                        popUpTekst="Sikker på at du vil slette kontonummer?"
                    >
                        Slett kontonummer
                    </KnappMedBekreftPopup>
                </FormKnapperWrapper>
            );
        const norskEllerUtenlandskKontoRadio = (
            <RadioPanelGruppe
                radios={this.radioKnappProps()}
                legend={''}
                name={'Velg norsk eller utenlandsk konto'}
                checked={this.state.norskKontoRadio ? bankEnum.erNorsk : bankEnum.erUtenlandsk}
                onChange={this.handleRadioChange}
            />
        );
        const kontoInputs = this.state.norskKontoRadio ? (
            this.getNorskKontonrInputs()
        ) : (
            <UtenlandskKontonrInputs
                bankkonto={this.state.bankkontoInput}
                bankkontoValidering={this.state.bankkontoValidering}
                updateBankkontoInputsState={this.updateBankkontoInputsState}
            />
        );
        const knapper = (
            <FormKnapperWrapper>
                <KnappBase
                    type="standard"
                    onClick={this.tilbakestill}
                    disabled={!this.formErEndret() || this.kontonummerBleLagret() || this.requestIsPending()}
                >
                    Avbryt
                </KnappBase>
                <KnappBase
                    type="hoved"
                    spinner={this.props.resourceStatus === STATUS.LOADING}
                    autoDisableVedSpinner={true}
                    disabled={
                        !this.formErEndret() ||
                        !veilederHarPåkrevdRolleForEndreKontonummer(this.props.veilederRoller) ||
                        this.kontonummerBleLagret()
                    }
                >
                    Endre kontonummer
                </KnappBase>
            </FormKnapperWrapper>
        );
        const endreKontonummerRequestTilbakemelding = (
            <RequestTilbakemelding
                status={this.props.resourceStatus}
                onError={'Det skjedde en feil ved endring av kontonummer.'}
                onSuccess={`Kontonummer ble endret.`}
            />
        );
        const konto = (
            <Undertekst>
                Gjeldende kontonummer:
                <FormatertKontonummer
                    kontonummer={(this.props.person.bankkonto && this.props.person.bankkonto.kontonummer) || ''}
                />
            </Undertekst>
        );

        const sistEndretInfo = <EtikettGrå>{hentEndringstekst(this.props.person.bankkonto)}</EtikettGrå>;
        return (
            <form onSubmit={this.handleSubmit}>
                <FormFieldSet disabled={!veilederHarPåkrevdRolleForEndreKontonummer(this.props.veilederRoller)}>
                    <Undertittel>Kontonummer</Undertittel>
                    {!erNyePersonoversikten() && konto}
                    {sistEndretInfo}
                    <EndreKontonummerInfomeldingWrapper veilderRoller={this.props.veilederRoller} />
                    {norskEllerUtenlandskKontoRadio}
                    {kontoInputs}
                    {knapper}
                    {sletteKnapp}
                    {endreKontonummerRequestTilbakemelding}
                </FormFieldSet>
            </form>
        );
    }
}

const mapStateToProps = (state: AppState): StateProps => {
    return {
        resourceStatus: state.restResources.endreKontonummer.status
    };
};

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        reloadPerson: (fødselsnummer: string) => dispatch(reloadPerson(fødselsnummer)),
        endreKontonummer: (fødselsnummer: string, request: EndreKontonummerRequest) =>
            dispatch(endreKontonummer(fødselsnummer, request)),
        resetEndreKontonummerResource: () => dispatch(reset())
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(EndreKontonummerForm);
