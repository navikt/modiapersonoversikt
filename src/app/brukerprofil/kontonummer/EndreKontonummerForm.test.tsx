import * as React from 'react';
import * as renderer from 'react-test-renderer';
import EndreKontonummerForm from './EndreKontonummerForm';
import TestProvider from '../../../test/Testprovider';
import { VeilederRoller } from '../../../models/veilederRoller';
import { aremark } from '../../../mock/person/aremark';
import { Bankkonto, Person } from '../../../models/person/person';

const veilerRoller: VeilederRoller = { roller: ['0000-GA-BD06_EndreKontonummer'] };

test('Viser riktig form for å endre norsk kontonummer', () => {
    const endrekontonummerForm = renderer.create(
        <TestProvider>
            <EndreKontonummerForm person={aremark} veilederRoller={veilerRoller} />
        </TestProvider>
    );

    const json = endrekontonummerForm.toJSON();
    expect(json).toMatchSnapshot();
});

test('Viser riktig form for å endre utenlandsk kontonummer', () => {
    const person: Person = {
        ...aremark,
        bankkonto: {
            ...(aremark.bankkonto as Bankkonto),
            landkode: {
                kodeRef: 'NRN',
                beskrivelse: 'Narnia'
            },
            valuta: {
                beskrivelse: 'Spesidaler',
                kodeRef: 'SPD'
            }
        }
    };

    const endrekontonummerForm = renderer.create(
        <TestProvider>
            <EndreKontonummerForm person={person} veilederRoller={veilerRoller} />
        </TestProvider>
    );

    const json = endrekontonummerForm.toJSON();
    expect(json).toMatchSnapshot();
});
