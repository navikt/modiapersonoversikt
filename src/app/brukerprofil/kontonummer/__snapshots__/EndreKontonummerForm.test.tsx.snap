// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser riktig form for å endre norsk kontonummer 1`] = `
.c2 {
  color: #78706a;
  margin-top: 3px;
}

.c2 .typo-etikett-liten {
  line-height: 1rem;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 > * {
  margin: 0.5rem 0 0.5rem 1em;
}

.c1 span:not(:last-child):after {
  content: '.';
}

<form
  onSubmit={[Function]}
>
  <fieldset
    className="c0"
    disabled={false}
  >
    <h2
      className="typo-undertittel"
    >
      Kontonummer
    </h2>
    <p
      className="typo-undertekst"
    >
      Gjeldende kontonummer:
      <span
        className="c1"
      >
        <span>
          1234
        </span>
        <span>
          56
        </span>
        <span>
          78911
        </span>
      </span>
    </p>
    <div
      className="c2"
    >
      <p
        className="typo-etikett-liten"
      >
        Endret 12.12.2005 z9910
      </p>
    </div>
    <div
      className="inputPanelGruppe"
    >
      <fieldset
        className="skjema__fieldset"
      >
        <legend
          className="skjema__legend"
        >
          
        </legend>
        <div
          className="inputPanelGruppe__inner"
        >
          <label
            className="inputPanel radioPanel inputPanel--checked"
            htmlFor="Kontonummer i Norge"
          >
            <input
              aria-checked={true}
              checked={true}
              className="inputPanel__field"
              id="Kontonummer i Norge"
              name="Velg norsk eller utenlandsk konto"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="Kontonummer i Norge"
            />
            <span
              className="inputPanel__label"
            >
              Kontonummer i Norge
            </span>
          </label>
          <label
            className="inputPanel radioPanel"
            htmlFor="Kontonummer i utlandet"
          >
            <input
              aria-checked={false}
              checked={false}
              className="inputPanel__field"
              id="Kontonummer i utlandet"
              name="Velg norsk eller utenlandsk konto"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="Kontonummer i utlandet"
            />
            <span
              className="inputPanel__label"
            >
              Kontonummer i utlandet
            </span>
          </label>
        </div>
      </fieldset>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Kontonummer
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="1234 56 78911"
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="c3"
    >
      <button
        className="knapp knapp--standard knapp--disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Avbryt
      </button>
      <button
        className="knapp knapp--hoved knapp--disabled"
        disabled={true}
        type="submit"
      >
        Endre kontonummer
      </button>
    </div>
    <div
      className="c3"
    >
      <button
        className="knapp knapp--standard"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Slett kontonummer
      </button>
    </div>
  </fieldset>
</form>
`;

exports[`Viser riktig form for å endre utenlandsk kontonummer 1`] = `
.c2 {
  color: #78706a;
  margin-top: 3px;
}

.c2 .typo-etikett-liten {
  line-height: 1rem;
}

.c0 {
  margin: 0;
  padding: 0;
  border: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 > * {
  margin: 0.5rem 0 0.5rem 1em;
}

.c1 span:not(:last-child):after {
  content: '.';
}

<form
  onSubmit={[Function]}
>
  <fieldset
    className="c0"
    disabled={false}
  >
    <h2
      className="typo-undertittel"
    >
      Kontonummer
    </h2>
    <p
      className="typo-undertekst"
    >
      Gjeldende kontonummer:
      <span
        className="c1"
      >
        <span>
          1234
        </span>
        <span>
          56
        </span>
        <span>
          78911
        </span>
      </span>
    </p>
    <div
      className="c2"
    >
      <p
        className="typo-etikett-liten"
      >
        Endret 12.12.2005 z9910
      </p>
    </div>
    <div
      className="inputPanelGruppe"
    >
      <fieldset
        className="skjema__fieldset"
      >
        <legend
          className="skjema__legend"
        >
          
        </legend>
        <div
          className="inputPanelGruppe__inner"
        >
          <label
            className="inputPanel radioPanel"
            htmlFor="Kontonummer i Norge"
          >
            <input
              aria-checked={false}
              checked={false}
              className="inputPanel__field"
              id="Kontonummer i Norge"
              name="Velg norsk eller utenlandsk konto"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="Kontonummer i Norge"
            />
            <span
              className="inputPanel__label"
            >
              Kontonummer i Norge
            </span>
          </label>
          <label
            className="inputPanel radioPanel inputPanel--checked"
            htmlFor="Kontonummer i utlandet"
          >
            <input
              aria-checked={true}
              checked={true}
              className="inputPanel__field"
              id="Kontonummer i utlandet"
              name="Velg norsk eller utenlandsk konto"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="Kontonummer i utlandet"
            />
            <span
              className="inputPanel__label"
            >
              Kontonummer i utlandet
            </span>
          </label>
        </div>
      </fieldset>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Velg land
      </label>
      <div
        className="selectContainer input--fullbredde"
      >
        <select
          className="skjemaelement__input"
          disabled={false}
          id="Helt tilfeldig ID"
          onChange={[Function]}
          value="NRN"
        >
          <option
            disabled={true}
            value=""
          >
            Velg Land
          </option>
          <option
            value="BMU"
          >
            Bermuda
             (
            BMU
            )
          </option>
          <option
            value="COL"
          >
            Colombia
             (
            COL
            )
          </option>
          <option
            value="LVS"
          >
            Langtvekkistan
             (
            LVS
            )
          </option>
          <option
            value="MRS"
          >
            Mars
             (
            MRS
            )
          </option>
          <option
            value="NRN"
          >
            Narnia
             (
            NRN
            )
          </option>
          <option
            value="NOR"
          >
            Norge
             (
            NOR
            )
          </option>
          <option
            value="ESP"
          >
            Spania
             (
            ESP
            )
          </option>
          <option
            value="SYD"
          >
            Syden
             (
            SYD
            )
          </option>
          <option
            value="USA"
          >
            Usa
             (
            USA
            )
          </option>
          <option
            value="UTL"
          >
            Utlandet
             (
            UTL
            )
          </option>
        </select>
      </div>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Bankens navn
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="Bien sparebank"
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Bankens adresse
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        BC/SWIFT-kode
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Kontonummer eller IBAN
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="12345678911"
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Bankkode
      </label>
      <input
        className="skjemaelement__input input--fullbredde"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="skjemaelement"
    >
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Velg valuta
      </label>
      <div
        className="selectContainer input--fullbredde"
      >
        <select
          className="skjemaelement__input"
          disabled={false}
          id="Helt tilfeldig ID"
          onChange={[Function]}
          value="SPD"
        >
          <option
            disabled={true}
            value=""
          >
            Velg valuta
          </option>
          <option
            value="USD"
          >
            Amerikanske Dollar
             (
            USD
            )
          </option>
          <option
            value="GUL"
          >
            Gullmynter, Kong Salomons Rike
             (
            GUL
            )
          </option>
          <option
            value="NZD"
          >
            New Zealand Dollar
             (
            NZD
            )
          </option>
          <option
            value="NOK"
          >
            Norske Kroner
             (
            NOK
            )
          </option>
          <option
            value="SPD"
          >
            Spesidaler
             (
            SPD
            )
          </option>
        </select>
      </div>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="c3"
    >
      <button
        className="knapp knapp--standard knapp--disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Avbryt
      </button>
      <button
        className="knapp knapp--hoved knapp--disabled"
        disabled={true}
        type="submit"
      >
        Endre kontonummer
      </button>
    </div>
    <div
      className="c3"
    >
      <button
        className="knapp knapp--standard"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Slett kontonummer
      </button>
    </div>
  </fieldset>
</form>
`;
